crucible-llvm/src/Lang/Crucible/LLVM/SimpleLoopFixpoint.hs

```
@@ -0,0 +1,748 @@
+
+  verbSetting <- W4.getOptionSetting W4.verbosity $ W4.getConfiguration sym
+  verb <- fromInteger <$> W4.getOpt verbSetting
+
+  let flattenWTOComponent = \case
+        C.SCC C.SCCData{..} ->  wtoHead : concatMap flattenWTOComponent wtoComps
+        C.Vertex v -> [v]
+  let loop_map = Map.fromList $ mapMaybe
+        (\case
+          C.SCC C.SCCData{..} -> Just (wtoHead, wtoHead : concatMap flattenWTOComponent wtoComps)
+          C.Vertex{} -> Nothing)
+        (C.cfgWeakTopologicalOrdering cfg)
+
+  fixpoint_stuff_ref <- newIORef @(FixpointStuff sym blocks) $ FixpointStuff
+    { fixpointStatus = BeforeFixpoint
+    , fixpointBlockId = undefined
+    , fixpointAssumptionFrameIdentifier = undefined
+    , fixpointSubstitution = undefined
+    , fixpointRegMap = undefined
+    , fixpointMemSubstitution = undefined
+    , fixpointLoopCondition = undefined
+    , fixpointLoopIndexBound = undefined
+    }
+
+  return $ C.ExecutionFeature $ \exec_state -> do
+    let ?logMessage = \msg -> when (verb >= (3 :: Natural)) $ do
+          let h = C.printHandle $ C.execStateContext exec_state
+          System.IO.hPutStrLn h msg
+          System.IO.hFlush h
+    fixpoint_stuff <- readIORef fixpoint_stuff_ref
+    case exec_state of
+      C.RunningState (C.RunBlockStart block_id) sim_state
+        | C.SomeHandle cfgHandle == C.frameHandle (sim_state ^. C.stateCrucibleFrame)
+        , Just Refl <- W4.testEquality
+            (fmapFC C.blockInputs cfgBlockMap)
+            (fmapFC C.blockInputs $ C.frameBlockMap $ sim_state ^. C.stateCrucibleFrame)
+        , Map.member (C.Some block_id) loop_map -> case fixpointStatus fixpoint_stuff of
+
+          BeforeFixpoint -> do
+            ?logMessage $ "SimpleLoopFixpoint: RunningState: " ++ show BeforeFixpoint ++ " -> " ++ show ComputeFixpoint
+            assumption_frame_identifier <- C.pushAssumptionFrame sym
+            let mem_impl = fromJust $ C.lookupGlobal mem_var (sim_state ^. C.stateGlobals)
```

Can we make this a panic if the variable is missing? At a minimum, a `HasCallStack` constraint would make this easier to track down if messed up